### PR TITLE
Convert Ui-Master Package to Js

### PR DESCRIPTION
**Pull request by MartinSchoeler**
@RocketChat/core

### CODE REVIEW

File: packages/rocketchat-ui-master/client/main.js [hunk 1]
```diff
@@ -0,0 +1,303 @@
+/* globals toolbarSearch, menu, isRtl, fireGlobalEvent, CachedChatSubscription */
+Template.body.onRendered(function() {
+	$(document.body).on('keydown', function(e) {
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: What happened with `clipboard`?

File: packages/rocketchat-ui-master/client/main.js [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+					fields: {
+						unread: 1,
+						alert: 1,
+						rid: 1,
+						t: 1,
+						name: 1,
+						ls: 1
+					}
+				});
+				fetch = subscriptions.fetch();
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: Where did you declare this var?

File: packages/rocketchat-ui-master/client/main.js [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+				const results = [];
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: There is no need to get and return results here

File: packages/rocketchat-ui-master/client/main.js [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+				Object.keys(fetch).forEach((key) =>{
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: The fetch's result is not an object, it's an array

File: packages/rocketchat-ui-master/client/main.js [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+		const ready = (Meteor.userId() == null) || (routerReady && subscriptionsReady);
+		RocketChat.CachedCollectionManager.syncEnabled = ready;
+		return ready;
+	},
+	hasUsername() {
+		return (Meteor.userId() != null) && (Meteor.user().username != null);
+	},
+	requirePasswordChange() {
+		const user = Meteor.user();
+		return (user ? user.requirePasswordChange : null) === true;
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: You can simplify:
```
return user && user.requirePasswordChange === true;
```

File: packages/rocketchat-ui-master/client/main.js [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+			console.log('room click .burger');
+		}
+		return menu.toggle();
+	},
+	'touchstart'(e, t) {
+		if (document.body.clientWidth > 780) {
+			return;
+		}
+		t.touchstartX = null;
+		t.touchstartY = null;
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: Why did you change from `undefined` to `null`?

File: packages/rocketchat-ui-master/client/main.js [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+		if ($(e.currentTarget).closest('.main-content').length > 0) {
+			t.touchstartX = e.originalEvent.touches[0].clientX;
+			t.touchstartY = e.originalEvent.touches[0].clientY;
+			t.mainContent = $('.main-content');
+			return t.wrapper = $('.messages-box > .wrapper');
+		}
+	},
+	'touchmove'(e, t) {
+		if (t.touchstartX != null) {
+			const touch = e.originalEvent.touches[0];
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: You can use `const [touch] = e.originalEvent.touches;` here

File: packages/rocketchat-ui-master/client/main.js [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+				return $('.input-message').focus();
+			}
+		}, 100);
+	});
+	return Tracker.autorun(function() {
+		swal.setDefaults({
+			cancelButtonText: t('Cancel')
+		});
+		const user = Meteor.user();
+		const settings = user.settings;
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: `user` can be undefined

File: packages/rocketchat-ui-master/client/main.js [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+		const prefs = user ? settings ? settings.preferences : null : null;
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: Simplify with `const prefs = settings && settings.preferences`

File: packages/rocketchat-ui-master/client/main.js [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+		if (prefs != null ? prefs.hideUsernames : null) {
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: Simplify with `if (prefs && prefs.hideUsernames) {`

File: packages/rocketchat-ui-master/client/main.js (hunk 1 above)
Comment (MartinSchoeler) [inline packages/rocketchat-ui-master/client/main.js]: It was never used again, breaking a eslint rule

Comment (rodrigok) [inline packages/rocketchat-ui-master/client/main.js]: Ok, so remove the variable assign, but keep the Clipboard instantiation
